base/bundle.hpp

```diff
@@ -0,0 +1,140 @@
+#pragma once
+
+#include <experimental/optional>
+#include <functional>
+#include <list>
+#include <memory>
+#include <mutex>
+#include <vector>
+
+#include "base/macros.hpp"
+#include "base/not_null.hpp"
+#include "base/status.hpp"
+
+namespace principia {
+namespace base {
+
+// Functions that can be cooperatively aborted should check |AbortRequested()|
+// when appropriate.
+extern thread_local std::function<bool()> AbortRequested;
```

Comment:
Could we make this a static member of the bundle, it seems disconnected here?

Also, the comment should mention that this should exclusively be called from the `task_body`.

This doesn't really work when `task_body` itself starts threads.  I think that I preferred it when this was a member function of the `Bundle` because then it was clear at the call site what was being checked.